js/web/lib/wasm/jsep/webgpu/ops/common.ts

```diff
@@ -805,4 +805,4 @@ export const getBroadcastDims = (inShape: readonly number[], outShape: readonly
 };
 
 // TODO: remove this limitation once >4D dims are supported by uniform.
-export const enableShapesUniforms = (rank: number): boolean => rank <= 4;
+export const enableShapesUniforms = (rank: number): boolean => rank <= 4 && rank > 0;
```

Comment:
I prefer you remove `&& rank > 0` here. And in binary-op.ts. Do the check like below. I think rank = 0 belongs to the uniform scope.
```
          inputDependencies: useShapesUniforms ?
             ((vectorize && !doBroadcast) || isAOneElement || isBOneElement ? ['type', 'type'] : ['rank', 'rank']) :
              ['dims', 'dims'],
        },

Reply:
Yeah. I was trying to suppress creating unnecessary variable using a condition involving vectorize and doBrodcast.

Reply:
I think there is more work to be done in the backend_webgpu.ts code to make uniforms work for scalars independent of binary ops ie. handling dims.length === 0 case.

Reply:
Do it work if you do below changes in backend_webtpu.ts?
```
--- a/js/web/lib/wasm/jsep/backend-webgpu.ts
+++ b/js/web/lib/wasm/jsep/backend-webgpu.ts
@@ -344,7 +344,7 @@ export class WebGpuBackend {
       const offsets: number[] = [];
       let maxAlignmentOfField = 1;
       programUniforms.forEach(v => {
-        const data = typeof v.data === 'number' ? [v.data] : v.data;
+        const data = typeof v.data === 'number' ? [v.data] : v.data.length === 0 ? [1] : v.data;
         // https://www.w3.org/TR/WGSL/#alignof

Reply:
This change combined with the previous suggested change resulted in 36 failures in suit0 node test. I think `dims.length===0` corner case needs to be handled separately. Similarly suppressing unnecessary/unused variables needs to be handled in common.ts separately. This issue exists independent of uniforms.

Reply:
It seems that @axinging met the similar issue with you. I'd like the fix like this https://github.com/microsoft/onnxruntime/commit/feb85da686e2ebacb0a4b6954880e1ff38bb7c6f.